packages/sdk/src/utils/index.ts

```diff
@@ -0,0 +1,360 @@
+): Signer {
+  const ECPair = ECPairFactory(ecc);
+
+  let privateKey: Uint8Array | undefined = signer.privateKey!;
+  if (!privateKey) {
+    throw new OrditSDKError("Private key is required for tweaking signer!");
+  }
+  if (signer.publicKey[0] === 3) {
+    privateKey = ecc.privateNegate(privateKey);
+  }
+
+  const tweakedPrivateKey = ecc.privateAdd(
+    privateKey,
+    tapTweakHash(toXOnly(signer.publicKey), opts.tweakHash),
+  );
+  if (!tweakedPrivateKey) {
+    throw new OrditSDKError("Invalid tweaked private key!");
+  }
+
+  return ECPair.fromPrivateKey(Buffer.from(tweakedPrivateKey), {
+    network: opts.network,
+  });
+}
+
+export const isObject = (o: unknown) => o?.constructor === Object;
+export const isString = (s: unknown) =>
+  s instanceof String || typeof s === "string";
+
+function encodeDecodeObject(
+  obj: NestedObject,
+  { encode, depth = 0 }: EncodeDecodeObjectOptions,
+) {
+  const maxDepth = 5;
+
+  if (depth > maxDepth) {
+    throw new OrditSDKError("Object too deep");
+  }
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const key in obj) {
+    // eslint-disable-next-line no-prototype-builtins, no-continue
+    if (!obj.hasOwnProperty(key)) continue;
+
+    const value = obj[key];
+    if (isObject(value)) {
+      // eslint-disable-next-line no-param-reassign
+      obj[key] = encodeDecodeObject(value as NestedObject, {
+        encode,
+        depth: depth + 1,
+      });
+    } else if (isString(value)) {
+      // eslint-disable-next-line no-param-reassign
+      obj[key] = encode
+        ? encodeURIComponent(value as string)
+        : decodeURIComponent(value as string);
+    }
+  }
+
+  return obj;
+}
+
+// TODO: Make this function immutable
+/**
+ * Encodes an object into [valid URI components](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent),
+ * modifying the object in the process.
+ *
+ * @param obj Object
+ * @returns The object is mutated.
+ * @deprecated
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function UNSTABLE_encodeObject(obj: NestedObject) {
+  return encodeDecodeObject(obj, { encode: true });
+}
+
+// TODO: Make this function immutable
+/**
+ * Decodes an object into [valid URI components](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent),
+ * modifying the object in the process.
+ *
+ * @param obj Object
+ * @returns The object is mutated.
+ * @deprecated
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function UNSTABLE_decodeObject(obj: NestedObject) {
+  return encodeDecodeObject(obj, { encode: false });
+}
+
+// Temporary convertors until bignumber.js is integrated
+export function convertSatoshisToBTC(satoshis: number) {
+  return satoshis / 10 ** 8;
+}
+
+export function convertBTCToSatoshis(btc: number) {
+  return parseInt((btc * 10 ** 8).toString(), 10); // remove floating point overflow by parseInt
```

Comment:
Should we convert it now to BigNumber?

Reply:
I'll do this in the 3rd PR (after this and PSBTBuilder are merged). This function is not used here yet.